controllers/component_controller.go

```diff
@@ -438,10 +477,22 @@ func (r *ComponentReconciler) generateGitops(ctx context.Context, req ctrl.Reque
 		return fmt.Errorf("unable to create temp directory for GitOps resources due to error: %v", err)
 	}
 
+	deployAssociatedComponents, err := devfileParser.GetDeployComponents(compDevfileData)
```

Comment:
Making a mental note here: I'll need to rebase after this has been merged to move this section into our gitops-generator go module/job.